frontend/server/utils.ts

```diff
@@ -65,6 +66,32 @@ export function loadJSON<T>(filepath?: string, defaultValue?: T): T | undefined
   }
 }
 
+/**
+ * prune remainingMaybePrunePath to relativePath, then concat with rootPath.
+ * for example:
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', undefined))
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', a))
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', a/b))
+ * console.log(pruneAndConcatPath('/data', 'a/b/c', other))
+ *
+ * result is:
+ * /data/a/b/c
+ * /data/b/c
+ * /data/c
+ * /data/a/b/c
```

Comment:
one example, one result is easier to read.

or you can also omit this part, because they are in the unit tests. Unit tests are documentation!